doc/developers/contributing.rst

```diff
@@ -603,6 +603,33 @@ Finally, follow the formatting rules below to make it consistently good:
     SelectKBest : Select features based on the k highest scores.
     SelectFpr : Select features based on a false positive rate test.
 
+* When documenting the parameters and attributes, here is a list of some
+  well-formatted examples::
+
+    n_clusters : int, default=3
+        The number of clusters detected by the algorithm.
+
+    some_param : {'hello', 'goodbye'}, bool or int, default=True
+        The parameter description goes here, which can be either a string
+        literal (either `hello` or `goodbye`), a bool, or an int. The default
+        value is True.
+
+    array_parameter : {array-like, sparse matrix, dataframe} of shape (n_samples, n_features) or (n_samples,)
```

Comment:
uhh did we have a discussion of "of shape" vs "shape=". Did someone count that?

Reply:
"of shape": 513 lines; "shape=" 338 lines; "shape =" 1316 lines. "of shape =": 162 lines, "of shape=" 1 line.

These are regex matches, so non-exclusive, so "of shape" includes "of shape =".
So we have 350 "of shape (...)" and 1115 "shape = (..)"

Reply:
Yes `shape=` would result in a smaller diff. Since this PR will result in many docstring changes, we should decide based on which is subjectively better.

(I have been +0.25 for `shape=` because it saves a few more characters.) Do you have a opinion on this?

Reply:
My preference is shape= but I don't mind that much.

Reply:
we are often short on space in that line.

``array-like of shape (n_samples,)`` vs ``array-like, shape=(n_samples,)`` saves 2 characters. Not sure if that's a good argument?

Reply:
I was thinking of pushing it to:

```py
array-like shape=(n_samples,)
```

Extreme savings of 3 characters.

@glemaitre I know you prefer `of shape` because it is easier to parse. Would you be okay with `shape=`?

Reply:
I don't like not having the comma :P

Reply:
![image](https://user-images.githubusercontent.com/449558/62652933-77221c80-b92a-11e9-8105-49e2f2183cbd.png)

Reply:
I don't think saving 2-3 chars is a major improvement here. Yes, that avoids having to use `\` in a few cases, but the rendered html are what are important, and they don't care.

Reply:
> My preference is shape= but I don't mind that much.

I also prefer shape= a little bit, I merge this because I think it's not important as long as the whole repo is consistent.

Reply:
we can fix those cases that don't comply with this with a `sed`, so I don't think the number of instances of each case are important that much anyway. I kinda liked not having the comma cause the comma is separating other things:

`int, array-like of shape (n_samples,), or None, default=None` (note that we don't have the oxford comma in our guide now, but I rather have it)

Reply:
> I don't like not having the comma :P

I prefer `of shape` if there is no comma. I find it more difficult to visually parse it.
I also find it closer to literal English --- even if we are writing documentation for expert computer scientists (I should not troll about Matlab post).

Anyway, I am going with the consensus (since it is merged it seems we are going for that one?)

Reply:
Ok, fine with keeping it.
@adrinjalali feel free to go ahead and do the sed and then fix all the merge conflicts :P
If whis is the way we want to go, we should probably do the sed.

Reply:
So are we doing this?

Reply:
I guess so, I'll submit a PR.